table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/CQLStashTableDAO.java

```diff
@@ -0,0 +1,307 @@
+package com.bazaarvoice.emodb.table.db.astyanax;
+
+import com.bazaarvoice.emodb.common.json.JsonHelper;
+import com.bazaarvoice.emodb.datacenter.api.DataCenters;
+import com.datastax.driver.core.BatchStatement;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.Iterators;
+import com.google.inject.Inject;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * This class is used to create and query a snapshot of tables for Stash.  For example, if a table is deleted, moved or
+ * has its metadata altered mid-Stash the output should still include the content as it would have appeared at the beginning
+ * of Stash.  It also provides an efficient API for querying which tables in that snapshot fall within a specific
+ * token range.
+ *
+ * Since this class and {@link com.bazaarvoice.emodb.table.db.TableDAO} both provide low level access to tables here's an
+ * explanation for why Stash-related table operations have been split from that class.  First, it's cleaner to separate
+ * the concerns for Stash from the needs of general table management, which are quite complex in their own right.
+ * Furthermore, TableDAO is used for both system of record and blob tables while Stash only applies to the former, so
+ * keeping the Stash portion here reinforces this separation.  For these reasons TableDAO focuses on general table management
+ * and delegates Stash operations to this class.  Consequently this class works exclusively with the serializable form of
+ * tables, {@link TableJson}, leaving the domain knowledge of converting these back into tables to TableDAO.
+ */
+public class CQLStashTableDAO {
+    
+    private final static String STASH_TOKEN_RANGE_TABLE = "stash_token_range";
+    // No Stash run should take over 1 day, so set the TTL to clean up stash tables if they aren't explicitly cleaned by then,
+    // plus a small buffer.
+    private final static int TTL = (int) TimeUnit.HOURS.toSeconds(28);
```

Comment:
We do explicitly delete the records in `LocalScanUploadMonitor#cleanupScan(String)`.  The TTL is just there as a backup in case something happens and the stash never finishes cleanly.

I have it 28 hours under the assumption that a Stash run should take no more than a day, so it already has a 4 hour buffer built in.  I can make it 3 days to give even more buffer, but if Stash takes over a day it's probably useless anyway.